### ISSUE TITLE
mmcv RoIAlign mix every rois while torchvision version works correctly

### ISSUE THREAD

**Issue by Hyunseok-Kim0** [open-mmlab/mmcv#2203]
**Checklist**

1. I have searched related issues but cannot get the expected help.
2. I have read the [FAQ documentation](https://mmcv.readthedocs.io/en/latest/trouble_shooting.html) but cannot get the expected help.
3. The unexpected results still exist in the latest version.

**Describe the Issue**

The sample code stated below is dividing a video frames into four parts. when use_torchvision flag of mmcv RoIAlign is set, the result looks fine. However, when use_torchvision is False, all of the roi results are fused together and looks very weird.
I also have tried to use native torchvision roi_align operations, which works correctly. I guess something in mmcv not working correctly.

![Screenshot from 2022-08-17 15-20-51](https://user-images.githubusercontent.com/34959032/185049767-952c7022-f768-44e5-96cf-ae1f06920210.png)
result when use_torchvision = True

![Screenshot from 2022-08-17 15-21-55](https://user-images.githubusercontent.com/34959032/185049780-a68d62af-70cf-45ea-97fe-57e92b958f4a.png)
result when use_torchvision = False

**Reproduction**

1. What command, code, or script did you run?

        import torch
        import cv2
        from torch import nn
        from einops import rearrange
        import numpy as np
        from mmcv.ops import RoIAlign
        
        
        class roi_test_model(nn.Module):
        
            def __init__(self):
                super().__init__()
        
                output_size = (112, 112)
                spatial_scale = 1.0
                pool_mode = "avg"
                sampling_ratio = 0
                aligned = True
        
                self.roi_align = RoIAlign(
                    output_size,
                    spatial_scale,
                    sampling_ratio=sampling_ratio,
                    pool_mode=pool_mode,
                    aligned=aligned, use_torchvision=True)
        
            def forward(self, x, roi):
        
                return self.roi_align(x, roi)
        
        
        def main():
            model = roi_test_model()
            model.eval()
            model.cuda()
        
            video = "marching.mp4"
            cap = cv2.VideoCapture(video)
            read = True
        
            # x1, y1, x2, y2
            dummy_roi = torch.Tensor([[0, 0, 0, 112, 112],
                                     [0, 112, 0, 224, 112],
                                     [0, 0, 112, 112, 224],
                                     [0, 112, 112, 224, 224]])
        
            dummy_roi.cuda()
        
            while read:
                read, frame = cap.read()
                # frame = cv2.cvtColor(frame, cv2.COLOR_BGR2GRAY)
                # frame = np.expand_dims(frame, axis=-1)
                image = frame.copy()
                if not read:
                    break
        
                frame = torch.Tensor(cv2.resize(frame, (224, 224)))
                frame = torch.unsqueeze(frame, 0)
                # frame = torch.unsqueeze(frame, -1)
                frame = rearrange(frame, "n h w c -> n c h w")
                frame.cuda()
                with torch.no_grad():
                    roi_aligned = model(frame, dummy_roi)
        
                for i, aligned in enumerate(roi_aligned):
                    aligned = rearrange(aligned, "c h w -> h w c").cpu().numpy().astype(np.uint8)
                    cv2.imshow(f"aligned {i}", cv2.resize(aligned, (224, 224)))
        
                cv2.imshow("image", image)
                cv2.waitKey(10)
        
        
        if __name__ == "__main__":
            main()

3. Did you make any modifications on the code? Did you understand what you have modified?
I didn't modify any mmcv code.

**Environment**

1. Please run `python -c "from mmcv.utils import collect_env; print(collect_env())"` to collect necessary environment information and paste it here.
{'sys.platform': 'linux', 'Python': '3.7.13 (default, Mar 29 2022, 02:18:16) [GCC 7.5.0]', 'CUDA available': True, 'GPU 0,1,2,3': 'NVIDIA GeForce RTX 3090', 'CUDA_HOME': '/usr/local/cuda', 'NVCC': 'Cuda compilation tools, release 11.3, V11.3.109', 'GCC': 'gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0', 'PyTorch': '1.12.0', 'PyTorch compiling details': 'PyTorch built with:\n  - GCC 9.3\n  - C++ Version: 201402\n  - Intel(R) oneAPI Math Kernel Library Version 2021.4-Product Build 20210904 for Intel(R) 64 architecture applications\n  - Intel(R) MKL-DNN v2.6.0 (Git Hash 52b5f107dd9cf10910aaa19cb47f3abf9b349815)\n  - OpenMP 201511 (a.k.a. OpenMP 4.5)\n  - LAPACK is enabled (usually provided by MKL)\n  - NNPACK is enabled\n  - CPU capability usage: AVX2\n  - CUDA Runtime 11.3\n  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;-gencode;arch=compute_37,code=compute_37\n  - CuDNN 8.3.2  (built against CUDA 11.5)\n  - Magma 2.5.2\n  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.3, CUDNN_VERSION=8.3.2, CXX_COMPILER=/opt/rh/devtoolset-9/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.12.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=OFF, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_ROCM=OFF, \n', 'TorchVision': '0.13.0', 'OpenCV': '4.5.5', 'MMCV': '1.6.1', 'MMCV Compiler': 'GCC 7.5', 'MMCV CUDA Compiler': '11.3'}

2. You may add addition that may be helpful for locating the problem, such as
   - How you installed PyTorch \[e.g., pip, conda, source\]
I used pytorch 1.12 docker uploaded in docker hub.

**Comment by Hyunseok-Kim0** [comment]
Is there any progress? here is the gist for reproduction of the problem.

https://colab.research.google.com/gist/Hyunseok-Kim0/c2c28e5890c24d4c8fd899af222be053/roialign.ipynb#scrollTo=VDYSlUL6k5eW

**Comment by Hyunseok-Kim0** [comment]
It's been almost two weeks, still there is no response..
I think it can cause severe performance drop since it does not extract appropriate rois.
mmcv RoIAlign is used in many other mmlab packages with default option use_torchvision=False like below.

https://github.com/open-mmlab/mmaction2/blob/ee932ba41dabde480469dcfc52d91d22d20b8dda/mmaction/models/roi_extractors/single_straight3d.py#L63-L78

Can anyone tell me if this is bug or not?

**Comment by Hyunseok-Kim0** [comment]
solved the problem. RoIAlign does work correctly only for contiguous tensors.
In mmcv 1.3.16, external module used in RoIAlign raises runtime error when input tensor is not contiguous, quiet helpful for debugging. 
However, in later version it does not raise any error.

**Comment by zhouzaida** [comment]
Hi @Hyunseok-Kim0 , sorry for my late reply, and thanks for your detailed analysis. Maybe we can simply force the input to be contiguous in  [RoIAlignFunction](https://github.com/open-mmlab/mmcv/blob/7f09a983c3dffb75146c4b18e404e386ad647e2d/mmcv/ops/roi_align.py#L67).
